lib/elixir/lib/macro.ex

```diff
@@ -276,13 +276,21 @@ defmodule Macro do
     {call, line, List.insert_at([], integer, expr)}
   end
 
-  def pipe(expr, {call, line, args} = call_args, integer) when is_list(args) do
-    if is_atom(call) and Identifier.binary_op(call) != :error do
-      raise ArgumentError,
-            "cannot pipe #{to_string(expr)} into #{to_string(call_args)}, " <>
-              "the #{to_string(call)} operator can only take two arguments"
-    else
-      {call, line, List.insert_at(args, integer, expr)}
+  def pipe(expr, {op, line, args} = op_args, integer) when is_list(args) do
```

Comment:
I think the variable name `op` is misleading as it could be an `op` or a `call`. Not sure what to do.